src/Symfony/Component/Scheduler/Scheduler.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Scheduler;
+
+use Symfony\Component\Scheduler\Generator\MessageGenerator;
+
+/**
+ * @experimental
+ */
+final class Scheduler
+{
+    /**
+     * @var array<MessageGenerator>
+     */
+    private array $generators = [];
+    private int $index = 0;
+
+    /**
+     * @param iterator<Schedule> $schedules
+     */
+    public function __construct(private array $handlers, array $schedules) {
+        foreach ($schedules as $schedule) {
+            $this->addSchedule($schedule);
+        }
+    }
+
+    public function addSchedule(Schedule $schedule)
+    {
+        $this->addMessageGenerator(new MessageGenerator($schedule, 'schedule_'.$this->index++));
+    }
+
+    public function addMessageGenerator(MessageGenerator $generator)
+    {
+        $this->generators[] = $generator;
+    }
+
+    /**
+     * Schedules messages.
+     *
+     * Valid options are:
+     *  * sleep (default: 1000000): Time in microseconds to sleep after no messages are found
+     */
+    public function run(array $options = []): void
+    {
+        $options = array_merge([
```

Comment:
`$options += ['sleep' => 1000000];`

what about allowing an optional clock to the constructor ? :)
this time it should be a ClockInterface from Symfony, so that we can use it to sleep

Reply:
Done